test/files/run/t6488.scala

```diff
@@ -39,9 +39,10 @@ object Test {
     val reading = new CountDownLatch(1)
     val count   = new AtomicInteger
     def counted = count.get
-    val command = s"${f.getAbsolutePath} -classpath ${javaClassPath} Test data"
+    val outdir  = s"$userDir/test/files/run/t6488-run.obj"
+    val command = s"${f.getAbsolutePath} -classpath ${javaClassPath}${pathSeparator}${outdir} Test data"
```

Comment:
No idea. I forgot my Indiana Jones archaeology hat. I could investigate when the windows build started failing.

Reply:
Ah I didn't get the sequence of events. The broken test on windows made you discover the underlying bug. After fixing that, you still fix the test.

Though on non-windows, the manual outdir is not necessary right? Maybe there's a bug in `scala.util.Properties.javaClassPath`? Maybe the test should run in a forked vm?

Reply:
Actually, I still don't know. I'd assumed it was the `/bin/ls` fallback exec that worked on linux, but no, it should have run java with a `no such class` output. I recall that windows will block if i/o streams aren't consumed, was that the platform difference?

Reply:
Guess what `println(f)` shows? `C:\Program Files\Java\jdk1.8.0_152\jre\bin\java-rmi.exe` :-) Anyway, [that's not the issue](https://stackoverflow.com/questions/25055466/what-is-java-rmi-exe).

The test out dir is missing from the classpath indeed, on linux and windows, so the test never reaches the `data()` part, but that doesn't really matter, as there just needs to be *some* output, and the process produces `Error: Could not find or load main class Test`.

I still don't see why it would hang on windows not on linux, maybe you do?

Reply:
Now that the Red Sox have clinched it, I'll find time to try it out on Windows. I bought a replacement Windows box a year ago to use for work, but I dread turning it on. I did wonder what exe the test was finding.

My previous comment about windows process blocking if out or err is not consumed is due to this happening to me in a unit test in 2003. That might have been blocking due to hitting a buffer limit, though.

It would be nice to have a `words""` interpolator that uses `CommandLineParser` to split parts and takes args whole to render a `Seq[String]`.

Reply:
retronym mentioned caveats at the commit to run tests in-process: https://github.com/scala/scala/commit/3b935176bcd0ec22bbbe8317d0dce3b63a88d8bc

(I missed the change at the old repo, so by now, I haven't yet thought about all the implications for tests.)

FSR, listing the obj directory fails when running in-process, after working around "partest.output" not set the same when in-process. However, forcing the test to fork makes it work.

It would be nicer if env was same for in/out-of-process. I'll try to get back to the mystery of why `.obj` is not listable, since compiler output is the same, it's not a virtual dir like in REPL, right?